calyx/src/lang/pretty_print.rs

```diff
@@ -236,100 +235,103 @@ impl PrettyPrint for Control {
             .append(RcDoc::line())
             .append(self.tbranch.prettify(&arena))
             .append(RcDoc::line())
             .append(self.fbranch.prettify(&arena))
-            .nest(1);
-        parens(inner)
+            .nest(1)
+            .parens()
     }
 }
 
 impl PrettyPrint for While {
     fn prettify<'a>(&self, arena: &'a bumpalo::Bump) -> RcDoc<'a, ColorSpec> {
-        let inner = colors::control(RcDoc::text("while"))
+        RcDoc::text("while")
+            .control()
             .append(RcDoc::space())
             .append(self.port.prettify(&arena))
             .append(RcDoc::space())
-            .append(parens(small_vec(&self.cond, &arena)))
+            .append(small_vec(&self.cond, &arena).parens())
             .append(RcDoc::line())
             .append(self.body.prettify(&arena))
-            .nest(1);
-        parens(inner)
+            .nest(1)
+            .parens()
     }
 }
 
 impl PrettyPrint for Print {
     fn prettify<'a>(&self, arena: &'a bumpalo::Bump) -> RcDoc<'a, ColorSpec> {
-        let inner = colors::enable(RcDoc::text("print"))
+        RcDoc::text("print")
+            .control()
             .append(RcDoc::line())
             .append(self.var.prettify(&arena))
-            .group();
-        parens(inner)
+            .group()
+            .parens()
     }
 }
 
 impl PrettyPrint for Enable {
     fn prettify<'a>(&self, arena: &'a bumpalo::Bump) -> RcDoc<'a, ColorSpec> {
-        let inner = colors::enable(RcDoc::text("enable"))
+        RcDoc::text("enable")
+            .enable()
             .append(RcDoc::line())
             .append(self.comps.prettify(&arena))
-            .group();
-        parens(inner)
+            .group()
+            .parens()
     }
 }
 
 impl PrettyPrint for Empty {
     fn prettify<'a>(&self, _arena: &'a bumpalo::Bump) -> RcDoc<'a, ColorSpec> {
-        let inner = colors::enable(RcDoc::text("empty"));
-        parens(inner)
+        RcDoc::text("empty").control().parens()
     }
 }
 
 impl PrettyPrint for Port {
     fn prettify<'a>(&self, arena: &'a bumpalo::Bump) -> RcDoc<'a, ColorSpec> {
         match self {
-            Port::Comp { component, port: p } => {
-                let inner = colors::port(RcDoc::text("@"))
-                    .append(RcDoc::space())
-                    .append(component.prettify(&arena))
-                    .append(RcDoc::space())
-                    .append(p.prettify(&arena));
-                parens(inner)
-            }
-            Port::This { port: p } => {
-                let inner = colors::port(RcDoc::text("@"))
-                    .append(RcDoc::space())
-                    .append(colors::keyword(RcDoc::text("this")))
-                    .append(RcDoc::space())
-                    .append(p.prettify(&arena));
-                parens(inner)
-            }
+            Port::Comp { component, port: p } => RcDoc::text("@")
+                .port()
```

Comment:
What does this `port` call refer to? Is `port` a function on RcDoc?

Reply:
`port` is a method defined in the `ColorHelper` trait. This is assigning the "port" color to the `@` symbol. This can probably be fixed with better naming. But again, it's definitely a little weird to assign the color this way.

Reply:
Yeah, things that change the color should have a better name. Otherwise it reads like it is adding something called `port` to this component